src/sentry/mediators/sentry_app_installations/authorizer.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+
+import six
+
+from sentry.coreapi import APIUnauthorized
+from sentry.mediators import Mediator, Param
+from sentry.models import ApiGrant, ApiApplication, ApiToken, SentryApp
+from sentry.utils.cache import memoize
+
+
+class Authorizer(Mediator):
+    install = Param('sentry.models.SentryAppInstallation')
+    grant_type = Param(six.string_types)
+    code = Param(six.string_types)
+    client_id = Param(six.string_types)
+    user = Param('sentry.models.User')
+
+    def call(self):
+        self._validate_grant_type()
+        self._validate_install()
+        self._validate_sentry_app()
+        self._validate_grant()
+        return self.exchange()
+
+    def exchange(self):
+        return ApiToken.objects.create(
+            user=self.user,
+            application=self.application,
+            scope_list=self.sentry_app.scope_list,
+            refresh_token=None,
+            expires_at=None,
```

Comment:
Will there be any way for applications to rotate their keys if they never expire?